src/glfw/main-win32.cpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (C) 2022 The Antares Authors
+//
+// This file is part of Antares, a tactical space combat game.
+//
+// Antares is free software: you can redistribute it and/or modify it
+// under the terms of the Lesser GNU General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Antares is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with Antares.  If not, see http://www.gnu.org/licenses/
+
+#if defined(_WIN32) && defined(WIN32_NO_CONSOLE)
+
+#include <windows.h>
+#include <shellapi.h>
+#include <vector>
+#include <pn/string>
+
+int main(int argc, char* const* argv);
+
+int APIENTRY WinMain(HINSTANCE h_inst, HINSTANCE h_inst_prev, PSTR cmd_line, int cmd_show) {
+    LPWSTR* wargv;
+    int     argc;
+
+    wargv = CommandLineToArgvW(GetCommandLineW(), &argc);
+    if (!wargv) {
+        return -1;
+    }
+
+    std::vector<std::string> arg_strings;
+    std::vector<char*>       argv_vector;
+
+    for (int i = 0; i < argc; i++) {
+        const wchar_t* warg = wargv[i];
+        arg_strings.push_back(pn::string(warg, wcslen(warg)).cpp_str());
```

Comment:
There’s supposed to be a [1-arg pn::string constructor][1]. Does it not work?

[1]: https://github.com/arescentral/procyon/blob/924e1dd54e0d181868d3203ba1233b7017589405/src/cpp/include/pn/string#L216-L217